tools/github-event-processor/Azure.Sdk.Tools.GitHubEventProcessor/Utils/LabelUtils.cs

```diff
@@ -8,22 +9,31 @@ namespace Azure.Sdk.Tools.GitHubEventProcessor.Utils
     public class LabelUtils
     {
         /// <summary>
-        /// Check whether or not an label exists in a list of labels. This will be used on issues,
-        /// and PRs, both of which contain a list of labels
+        /// Overload that takes the Octokit's IReadOnlyList of Labels and converts it
+        /// to a string list.
         /// </summary>
         /// <param name="labels">IReadOnlyList of labels</param>
         /// <param name="labelToCheck">The label to look for on the issue</param>
         /// <returns>true if the label exists, false otherwise</returns>
         public static bool HasLabel(IReadOnlyList<Label> labels, string labelToCheck)
+        {
+            return HasLabel(labels.Select(l => l.Name).ToList(), labelToCheck);
+        }
+
+        /// <summary>
+        /// Check whether or not an label exists in a list of labels. This will be used on issues,
+        /// and PRs, both of which contain a list of labels
+        /// </summary>
+        /// <param name="labels">List of labels</param>
+        /// <param name="labelToCheck">The label to look for in the list</param>
+        /// <returns>true if the label exists, false otherwise</returns>
+        public static bool HasLabel(List<string> labels, string labelToCheck)
         {
             if (labels != null)
             {
-                foreach (Label label in labels)
+                if (labels.Contains(labelToCheck, StringComparer.OrdinalIgnoreCase))
```

Comment:
You could simplify this to:

```csharp
return labels.Contains(labelToCheck, StringComparer.OrdinalIgnoreCase);
```

Not much of a savings, but it allows you to skip the branch condition.

Reply:
Yes, but the false case is taken care of with the fall through which happens if labels is null.